src/btree/bt_curnext.c

```diff
@@ -793,25 +830,25 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
                 total_skipped += skipped;
                 break;
             case WT_PAGE_ROW_LEAF:
-                ret = __cursor_row_next(
-                  cbt, newpage, restart, &skipped, prefix, &prefix_key_out_of_bounds);
+                ret =
+                  __cursor_row_next(cbt, newpage, restart, &skipped, prefix, &key_out_of_bounds);
                 total_skipped += skipped;
-                /*
-                 * If we are doing a search near with prefix key configured, we need to check if we
-                 * have exited the cursor row next function due to a prefix key mismatch. If so, we
-                 * can immediately return WT_NOTFOUND and we do not have to walk onto the next page.
-                 */
-                if (prefix_key_out_of_bounds) {
-                    WT_ASSERT(session, ret == WT_NOTFOUND);
-                    return (WT_NOTFOUND);
-                }
                 break;
             default:
                 WT_ERR(__wt_illegal_value(session, page->type));
             }
             if (ret != WT_NOTFOUND)
                 break;
 
+            /*
+             * If we are doing a search near with prefix key configured or the cursor has bounds
+             * set, we need to check if we have exited the next function due to a prefix key
+             * mismatch or the key is out of bounds. If so, we can immediately return WT_NOTFOUND
+             * and we do not have to walk onto the next page.
+             */
+            if (key_out_of_bounds)
```

Comment:
We aren't returning directly here so that the cursor will get properly reset before we return WT_NOTFOUND to `__wt_btcur_next`. I'll update the comment in the code here and in `bt_curprev` accordingly.

Reply:
But if we don't return, then after the loop it will traverse more pages, it is not just about the comment, I think logic has problem.

Reply:
Actually in this case, @bvpvamsikrishna , ret will be WT_NOTFOUND here, thus if we break here, it will directly exit out of the traversal logic. The break follows the traversal loop here.